Content/CustomHooks/Mechanics.NoBuild.cs

```diff
@@ -204,9 +330,46 @@ public class ProtectionWorld : ModSystem
 	{
 		public static List<Rectangle> ProtectedRegions = new();
 
+		private static readonly Dictionary<Point16, Ref<Rectangle>> RuntimeRegionsByPoint = new();
```

Comment:
They are used when the shrines are active. 
The build protection when active could be changed to not use this, but there is the possibility that this will be used for other stuff in the future.